workflow/controller/operator.go

```diff
@@ -1678,6 +1687,29 @@ func (woc *wfOperationCtx) executeTemplate(nodeName string, orgTmpl wfv1.Templat
 	return node, nil
 }
 
+// Checks if the template has exceeded its deadline
+func (woc *wfOperationCtx) checkTemplateTimeout(tmpl *wfv1.Template, node *wfv1.NodeStatus) (*time.Time, error) {
+	if node == nil {
+		return nil, nil
+	}
+
+	if tmpl.Timeout != "" {
+		tmplMaxDuration, err := parseStringToDuration(tmpl.Timeout)
```

Comment:
We should not accept values that do not have a unit like `"1234"`. I think the string should always be a duration string (e.g. "1h`). So with that, you can use time.Duration()

Reply:
added the validation